include/xdg/moab/direct_access.h

```diff
@@ -266,20 +266,33 @@ class MBDirectAccess {
       tx.clear();
       ty.clear();
       tz.clear();
+      first_vertices.clear();
     }
 
     //! \brief Set the coordinates of a vertex based on an index into a contiguous block of memory
     //! \param idx The index of the block of memory
     //! \param i The index of the vertex in the block of memory
     //! \param v The vertex to set the coordinates of
-    void set_coords(int idx, int i, xdg::Vertex& v) {
+    void set_coords(int i, xdg::Vertex& v) {
+      // determine the correct contiguous memory block index to use
+      int idx = 0;
+      auto fe = first_vertices[idx];
+      while(true) {
+        if (i - fe.first < fe.second) { break; }
+        idx++;
+        fe = first_vertices[idx];
+      }
```

Comment:
Is this the same?
```suggestion
      while(i - fe.first >= fe.second) {
        fe = first_vertices[++idx];
      }
```
(I have a personal bias against `while(true)` loops unless necessary... )